### ISSUE TITLE
[Enhancement] [General] Linux Support via Wine

### ISSUE THREAD

**Issue by taaderbe** [ChaosRecipeEnhancer/ChaosRecipeEnhancer#443]
Hi I really love this app.

But when I swtich to Linux I can't use it any longer, any chance  to get Linux version?

I can at least offer professionell testing, which I do since 20 years, if it's needed.

**Comment by HiMarioLopez** [comment]
Sadly no Linux version is in the works since we deliver the app as a windows executable package `.exe`

A cross-platform version would require some drastic rework, if not a complete rebuild of the app. I think it's something I want to do in the future, but it's hard to balance such large changes / rebuilds with my full time job haha

Right now this is a .NET Framework 4.8 WPF app, if we rebuild it as a .NET 6/7 MAUI app or maybe even try another framework like Avalonia we could attain cross-platform compatibility. I would take some time to even estimate how much effort a rebuild would take.

Even further from the original codebase would be to rework it into a JavaScript/TypeScript based app using something like Electron. But even more to change up as we wouldn't be able to re-use any of our logic written in C#.

So long winded answer, but I hope you understand the difficulties found at the moment with porting to Linux. Thank you for offering your assistance for testing purposes. I will definitely keep the community updated if a rebuild is something we can start to seriously consider.

**Comment by taaderbe** [comment]
Thanks, yes I fully understand that this is not "easy done". I would love to use it on linux, anyway still good job thanks for that. Keep going with good work.

**Comment by ZerkerEOD** [comment]
@taaderbe, I was able to get the newest version running on Linux. I am running  EndeavourOS (Archbased) so you may need to research some of the steps and convert them to the system you are running.

First I made sure wine was fully up to date (I initially had wine-staging installed which was running version 8.something. I upgraded using `yay -S wine` which installed wine version 9.5. After the installation, I installed .net 8.0.3 via the link on the homepage. Wine will pick it up easily to double click and install. Then I downloaded the .msi and installed the same way (double click and work through the installer). 

I then ran into an issue and after some searching it seemed to come down to some fonts causing issues. So I needed to install winetricks (`yay -S winetricks`) then run `winetricks corefonts`

After the fonts were installed it started right up. I am about to actually see if it runs in game and on that I am not as hopeful but will let you know,.

**Comment by ZerkerEOD** [comment]
With the above, I was able to get it to run. In the game, I wasn't able to use "Windowed Fullscreen" as it would not overlay over the game. I had to set it to use Windowed for it to over lay. I was then able to bring up the stash screen and edit it to what I needed. So right now it looks like it works with the above work around on Linux.

**Comment by HiMarioLopez** [comment]
Wow! That is amazing news. Thank you so much for testing this out!

**Comment by ZerkerEOD** [comment]
Yeah, looking at Wine's documentation it appears I needed at least version 9.01-rc to handle an issue with the user32.dll, other than that it appears to work well.

**Comment by BorjaLL** [comment]
how were you able to handle the log in?
I installed Lutris, winetricks etc and I cant get chrome to get the auth back to the app after I log in and click authorise

**Comment by ZerkerEOD** [comment]
> how were you able to handle the log in? I installed Lutris, winetricks etc and I cant get chrome to get the auth back to the app after I log in and click authorise

Are you using KDE? If so they have a known issue where a library changes the case of the authentication key. I can try to find it but basically you need to intercept the request and use that directly. There are walkthroughs online. I'll see if I can find one later.

**Comment by HiMarioLopez** [comment]
I do not have a lot of insight as a Windows user, but do want to add this to the discussion regarding the OAuth flow:

When the program is installed, it creates registry entries in the Windows Registry:
![image](https://github.com/user-attachments/assets/ffee5dd0-57e5-4cd5-af8b-adeecda22e4e)

(Ignore the `Debug-Dev`, `Debug-Final`, `Debug-Old`, `Debug-Release` entries, those are added by me manually) The `(Default)` entry is fully valid, though.

So whatever y'all are doing to make this work on Linux, you'll need an entry to route requests from a `chaosrecipe://` 'URL' to the application correctly (from the browser)

**Comment by HiMarioLopez** [comment]
Here's the URL that is generated for that routing:

`chaosrecipe://auth?code=<<REDACTED>>&state=<<REDACTED>>`
